crates/workspace/src/dock.rs

```
@@ -351,6 +359,20 @@ impl Dock {
             .position(|entry| entry.panel.remote_id() == Some(panel_id))
     }
 
+    pub fn activate_first_activatable_panel(
+        &mut self,
+        window: &mut Window,
+        cx: &mut Context<Self>,
+    ) {
+        if let Some(panel_ix) = self
+            .panel_entries
+            .iter()
+            .position(|entry| entry.panel.activatable(cx))
+        {
+            self.activate_panel(panel_ix, window, cx);
+        }
+    }
```

In the case where there is no activatable panel entry, this will currently silently fail. 

This will most likely rarely happen. However, open to change this to instead return a result which can be used to inform the user there is no panel to toggle.

---

Can we inline this into a single place that uses this, and do a `log::info` there if nothing enabled is found for the activation?

---

Not quite sure whether I understand you properly here. Inlining this in `workspace.rs` would require `panel_entries` to be public, which is currently not the case. 
We could possibly reduce this method to just return the found panel index (and an error if none is found) and activate the panel after handling the error in the workspace itself. Is that close to what you were referring to?